addons/web_tour/static/src/js/running_tour_action_helper.js

```diff
@@ -95,14 +95,19 @@ var RunningTourActionHelper = core.Class.extend({
                 bubbles: true,
             }));
         } else if (values.$element.is("select")) {
-            var $options = values.$element.children("option");
+            var $options = values.$element.find("option");
             $options.prop("selected", false).removeProp("selected");
             var $selectedOption = $options.filter(function () { return $(this).val() === text; });
             if ($selectedOption.length === 0) {
                 $selectedOption = $options.filter(function () { return $(this).text().trim() === text; });
             }
+            if ($selectedOption.length === 0 && text.startsWith('index ')) {
+                const index = parseInt(text.substring(6));
+                $selectedOption = $options.slice(index, index + 1);
+            }
             $selectedOption.prop("selected", true);
             this._click(values);
+            values.$element.trigger("input");
```

Comment:
Why is this needed?

Reply:
Because this is how the blog archive month `<select>`'s behavior is defined. https://github.com/odoo/odoo/blob/16.0/addons/website_blog/views/website_blog_components.xml#L60

Reply:
Fair, I don't know if it's obvious to anyone when reading the code that this line is about triggering our custom owl `oninput`.
I let you judge if it's worth a comment.

Reply:
Added a comment.